src/aiida/orm/implementation/storage_backend.py

```diff
@@ -304,6 +304,23 @@ def maintain(self, full: bool = False, dry_run: bool = False, **kwargs) -> None:
         :param dry_run: flag to only print the actions that would be taken without actually executing them.
         """
 
+    @abc.abstractmethod
```

Comment:
Making this an abstract method instead of implementing it with `raise NotImplementedError` _will_ break other storage plugins. However, I guess the only ones out there are mine in `aiida-s3` and I can take care of updating them. That being said, is there really a problem with making this concrete and raising the exception?

Reply:
Good point, will turn non-abstract.